__tests__/app.test.js

```diff
@@ -119,26 +119,89 @@ describe("GET /api/articles/:article_id/comments", () => {
                 expect(typeof comment.created_at).toBe("string")
                 expect(typeof comment.author).toBe("string")
                 expect(typeof comment.body).toBe("string")
-                expect(typeof comment.article_id).toBe("number")
+                expect(comment.article_id).toBe(1)
             }) 
             expect(body.comments).toBeSortedBy('created_at', {descending: true})
               
         })
     })
-    test("returns correct error if article_id doesnt exist", () => {
-        return request(app).get('/api/articles/1223/comments')
+    test("returns 200 error if article has no comments and empty array", () => {
+        return request(app).get('/api/articles/2/comments')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.comments).toEqual([])   
+        })
+    })
+
+    test("returns 404 error if article_id doesnt exist", () => {
+        return request(app).get('/api/articles/1337/comments')
         .expect(404)
         .then(({ body }) => {
-            expect(body.msg).toBe(`article with ID: 1223 does not exist`)   
+            expect(body.msg).toBe(`article_id with value 1337 does not exist in articles`)   
         })
     })
 
-    test("returns correct error if incorrect format given", () => {
+    test("returns 400 error if incorrect format given", () => {
         return request(app).get('/api/articles/dasads/comments')
         .expect(400)
         .then(({ body }) => {
             expect(body.msg).toBe("invalid article id format")   
         })
     })
 
+
 })
+
+describe("POST /api/articles/:article_id/comments", () => {
+    test("returns posted comment", () => {
+        return request(app).post('/api/articles/1/comments')
+        .send({
+            username: "lurker",
+            body: "this is a comment"
+        })
+        .expect(201)
+        .then(({ body }) => {
+            expect(body.comment).toMatchObject({ 
+                comment_id: 19,
+                body: 'this is a comment',
+                article_id: 1,
+                author: 'lurker',
+                votes: 0,
+            })
+        })
+    })
```

Comment:
What should happen if a request was made that had some of the required fields missing in the request body? i.e. no username or body properties. 🤔